### ISSUE TITLE
Incoperate statix?

### ISSUE THREAD

**Issue by Mic92** [jtojnar/nixpkgs-hammering#124]
@NerdyPepper build this handy [tool](https://github.com/NerdyPepper/statix).
I looks like nice extension to the already existing checks in nixpkgs-hammering.
Thoughts?

**Comment by jtojnar** [comment]
Sounds reasonable. Maybe we can do it the other way round since statix has
two of the things I want to implement here:

- The program is written in Rust (so that we do not need both Rust and
Python in the closure).
- Support of fixes

On the other hand, n-h seems to have more detailed error descriptions.

Unfortunately, I currently don't have the capacity to work on this. Any
help welcome, thought.

**Comment by nerdypepper** [comment]
>On the other hand, n-h seems to have more detailed error descriptions.

Yeah, statix currently does not describe/explain issues at all. I do have "offline explanation" on the roadmap though. I see n-h also has a bunch of tests, statix does not have any tests at the moment.

 >so that we do not need both Rust and Python in the closure
 
 I am not too familiar with nixpkgs-hammering's codebase, how many checks depend on python? Are they trivial enough to translate to Rust?  

 I'd definitely like to see unison in developer tools around Nix, my end goal is to have all of this reside in rnix-lsp, or atleast integrate with it.

**Comment by jtojnar** [comment]
The most of the checks are written in Nix so the Python basically just constructs Nix code, runs `nix-instantiate` and the compiled rust binaries, and then joins the result JSONs and formats them.

Should be pretty easy to port to rust https://github.com/jtojnar/nixpkgs-hammering/blob/a24b5537fab53dec6dc638576bff19606ad4a33b/tools/nixpkgs-hammer

**Comment by Mic92** [comment]
There is already some rust here anyway: https://github.com/jtojnar/nixpkgs-hammering/tree/a24b5537fab53dec6dc638576bff19606ad4a33b/rust-checks/src

**Comment by jtojnar** [comment]
Yeah, currently the checks are compiled into programs which output JSON results when run and are executed by the Python program. I want to have Rust do everything so it is easier to package in Nixpkgs.
